modules/ROOT/pages/provisioning-raspberry-pi4.adoc

```diff
@@ -0,0 +1,188 @@
+
+```
+dnf install rpi-imager
+```
+
+NOTE: This also works inside a `toolbx` container.
+
+If not on Fedora Linux you'll need to follow the documentation for obtaining the imager. Once you have the imager up and running (on Fedora you can run with `rpi-imager` on the command line). You'll see the imager application load:
+
+image::raspberry-pi-imager.png["Raspberry Pi Imager Application"]
+
+At this point you can follow https://www.raspberrypi.org/documentation/computers/raspberry-pi.html#updating-the-bootloader[the documentation] for how to create the disk and then update your Raspberry Pi 4.
+
+## Installing FCOS and Booting via U-Boot
+
+To run FCOS on a Raspberry Pi 4 via U-Boot the SD card or USB disk needs to be prepared on another system and then the disk moved to the RPi4. After writing FCOS to the disk a few more files will need to be copied in place on the EFI partition of the FCOS disk. Check out the https://www.raspberrypi.com/documentation/computers/configuration.html#boot-folder-contents[Raspberry Pi Documentation] to read more about what these files are for.
+
+In this case we can grab these files from the `uboot-images-armv8`, `bcm283x-firmware`, `bcm283x-overlays` RPMs from the Fedora Linux repositories. First download them and store them in a temporary directory on your system:
+
+```nohighlight
+RELEASE=34 # The target Fedora Release. Use the same one that current FCOS is based on.
+mkdir -p /tmp/RPi4boot/boot/efi/
+sudo dnf install -y --downloadonly --release=$RELEASE --forcearch=aarch64 --destdir=/tmp/RPi4boot/  uboot-images-armv8 bcm283x-firmware bcm283x-overlays
+```
+
+Now extract the contents of the RPMs and copy the proper `u-boot.bin` for the RPi4 into place:
+
+```nohighlight
+for rpm in /tmp/RPi4boot/*rpm; do rpm2cpio $rpm | sudo cpio -idv -D /tmp/RPi4boot/; done
+sudo mv /tmp/RPi4boot/usr/share/uboot/rpi_4/u-boot.bin /tmp/RPi4boot/boot/efi/rpi4-u-boot.bin
+```
+
+Run `coreos-installer` to install to the target disk. There are https://coreos.github.io/coreos-installer/getting-started/[various ways] to run `coreos-installer` and install to a target disk. We won't cover them all here, but this workflow most closely mirrors the xref:bare-metal.adoc#_installing_from_the_container["Installing from the container"] documentation.
+
+```nohighlight
+FCOSDISK=/dev/sdX
+sudo coreos-installer install --architecture=aarch64 -i config.ign $FCOSDISK
+```
+
+NOTE: Make sure you provide an xref:producing-ign.adoc[Ignition config] when you run `coreos-installer`.
+
+Now mount up the EFI partition and copy the files over:
+
+```nohighlight
+mkdir /tmp/FCOSEFIpart
+sudo mount ${FCOSDISK}2 /tmp/FCOSEFIpart
```

Comment:
> Are we planning to change away from the 2nd partition?

I don't think there are any plans, but we've worked hard to encourage users to access partitions by label and not number.

> Would mount -L EFI-SYSTEM be preferrable (though less accurate because labels aren't guaranteed to be unique)?

Right, that's the problem with it.  It might mount the ESP on the host.